locales/messages.json

```
@@ -1115,10 +1115,12 @@
   "_UnsupportedPort.comment": "An example of {package_name} is zlib.",
   "UnsupportedPortDependency": "- dependency {value} is not supported.",
   "_UnsupportedPortDependency.comment": "'{value}' is the name of a port dependency.",
-  "UnsupportedPortFeature": "{spec} is only supported on '{supports_expression}'",
-  "_UnsupportedPortFeature.comment": "An example of {spec} is zlib:x64-windows. An example of {supports_expression} is windows & !static.",
   "UnsupportedShortOptions": "short options are not supported: '{value}'",
   "_UnsupportedShortOptions.comment": "'{value}' is the short option given",
+  "UnsupportedSupportsExpression": "{spec} is only supported on '{supports_expression}'. This usually means that there are known build failures, or runtime problems, when building other platforms. To ignore this and attempt to build {spec} anyway, rerun vcpkg with `--allow-unsupported`.",
```

```suggestion
  "UnsupportedSupportsExpression": "{package_name} is only supported on '{supports_expression}'. This usually means that there are known build failures, or runtime problems, when building other platforms. To ignore this and attempt to build {package_name} anyway, rerun vcpkg with `--allow-unsupported`.",
```

`{spec}` implies triplet information (see comment). I think `{package_name}` is the closest we have without defining a new placeholder.

---

We are printing triplet information here; that's @dg0yt's comment :)

---

Package, features and triplets done right...
See the gdal example I added to https://github.com/microsoft/vcpkg-tool/pull/879#issuecomment-1411492640.
What I want is app.
~~~
$ ./vcpkg install gdal[*]:x64-linux
Computing installation plan...
gdal[aws-ec2-windows] cannot be installed:
atlmfc[core]:x64-linux is only supported on 'windows'
~~~

---

I agree that would be an improvement but that's more than the 10 minute "this should explain what we mean" fix I was going for here.